src/sentry/api/issue_search.py

```diff
@@ -22,77 +14,40 @@
 )
 from sentry.utils.compat import map
 
-
-class IssueSearchVisitor(SearchVisitor):
-    key_mappings = {
-        "assigned_to": ["assigned"],
-        "bookmarked_by": ["bookmarks"],
-        "subscribed_by": ["subscribed"],
-        "assigned_or_suggested": ["assigned_or_suggested"],
-        "first_release": ["first-release", "firstRelease"],
-        "first_seen": ["age", "firstSeen"],
-        "last_seen": ["lastSeen"],
-        "active_at": ["activeSince"],
-        # TODO: Special case this in the backends, since they currently rely
-        # on date_from and date_to explicitly
-        "date": ["event.timestamp"],
-        "times_seen": ["timesSeen"],
-        "sentry:dist": ["dist"],
-    }
-    numeric_keys = SearchVisitor.numeric_keys.union(["times_seen"])
-    date_keys = SearchVisitor.date_keys.union(["active_at", "date"])
-
-    @cached_property
-    def is_filter_translators(self):
-        is_filter_translators = {
-            "assigned": (SearchKey("unassigned"), SearchValue(False)),
-            "unassigned": (SearchKey("unassigned"), SearchValue(True)),
-            "for_review": (SearchKey("for_review"), SearchValue(True)),
-            "linked": (SearchKey("linked"), SearchValue(True)),
-            "unlinked": (SearchKey("linked"), SearchValue(False)),
-        }
-        for status_key, status_value in STATUS_QUERY_CHOICES.items():
-            is_filter_translators[status_key] = (SearchKey("status"), SearchValue(status_value))
-        return is_filter_translators
-
-    def visit_is_filter(self, node, children):
-        # the key is "is" here, which we don't need
-        negation, _, _, _, search_value = children
-
-        if search_value.raw_value.startswith("["):
-            raise InvalidSearchQuery('"in" syntax invalid for "is" search')
-
-        if search_value.raw_value not in self.is_filter_translators:
-            raise InvalidSearchQuery(
-                'Invalid value for "is" search, valid values are {}'.format(
-                    sorted(self.is_filter_translators.keys())
-                )
-            )
-
-        search_key, search_value = self.is_filter_translators[search_value.raw_value]
-
-        operator = "!=" if is_negated(negation) else "="
-
-        return SearchFilter(search_key, operator, search_value)
-
-    def visit_boolean_operator(self, node, children):
-        raise InvalidSearchQuery(
-            'Boolean statements containing "OR" or "AND" are not supported in this search'
-        )
+is_filter_translation = {
+    "assigned": ("unassigned", False),
+    "unassigned": ("unassigned", True),
+    "for_review": ("for_review", True),
+    "linked": ("linked", True),
+    "unlinked": ("linked", False),
+}
+for status_key, status_value in STATUS_QUERY_CHOICES.items():
+    is_filter_translation[status_key] = ("status", status_value)
+
+
+issue_search_config = SearchConfig(**default_config.__dict__)
```

Comment:
This is going to pass these values by reference right? So if we accidentally modified them we'd change the event search as well. Should be do a `deepcopy` here just to futureproof this?

Maybe we could have `SearchConfig.from(search_config)` to safely duplicate config settings

Reply:
Yeah let me actually verify, I was actually having the same thought second look at this

Reply:
I think we're gonna want to actually just use `asdict` on the dataclass instance here